### ISSUE TITLE
'navigate files and folders with single click'  only effective in jupyter lab

### ISSUE THREAD

**Issue by zjqzero** [jupyter/notebook#7538]
## Description
upgrade to 7.3.1，'navigate files and folders with single click' not checked, but it's still affecting the behavior.

This option only effective in jupyter lab.
 
![Image](https://github.com/user-attachments/assets/f8e6f03f-fef7-447b-aeb5-9cfb1bed36c4)

**Comment by andrii-i** [comment]
I have tested this locally and I can't reproduce this with notebook 7.3.1 on M1 MacBook Pro macos 15.1.1 ('navigate files and folders with single click' is respected). 

Could you please provide more information about the system you are using? 


```
- Operating System and version: <!-- e.g. Linux Ubuntu 21.04 -->
- Browser and version: <!-- e.g. Chrome 92 -->
<!-- Please note the Notebook version you are working with. You can find this in the Help -> About Jupyter Notebook menu option or by running `jupyter --version` from your terminal -->
- Jupyter Notebook version: <!-- e.g. 3.1.7 -->
```

**Comment by pythonic2020** [comment]
Using notebook 7.3.2 in a conda env on Win64 running Python 3.9, using only default notebook settings, I am getting the single-click behavior even though "Navigate files and directories with single click" setting is unchecked. This erroneous behavior is exactly that described by @zjqzero. There is also other bad behavior after closing a browser tab that contained an opened .ipynb file where the mouse cursor changes to a crossed-out circle with the filename of the .ipynb file just closed (see screenshot below). This bad cursor behavior disappears after clicking on notebook background. 

![Image](https://github.com/user-attachments/assets/22668d9c-4522-433a-8bb7-2e5f928e8b4b)

If the "Navigate files and directories with single click" setting is checked and then unchecked, the behavior returns to normal.

This looks like a serious bug to me, but is probably not hard to fix.

**Comment by jtpio** [comment]
Thanks all for the report.


> If the "Navigate files and directories with single click" setting is checked and then unchecked, the behavior returns to normal.

This can likely be the workaround for now.

Otherwise the issue mentioned above with files being dragged does seem to be easily reproducible indeed.

**Comment by miellaby** [comment]
@jtpio please note the single-click-causes-cursor-issue bug is worse than mentioned. In Firefox and Chrome, opening a file from the filebrowser (view single click mode) will open a new browser tab, but also hangs the filebrowser in drag&drop mode (this is the actual bug). When the end-user comes back to the filebrowser and clicks on a folder instead of the background, the filebrowser does move the opened file.

This bug is very annoying for non-techy users. I share locked-down jupyter notebooks with my coworkers as a form of rudimentary GUI, and they can't help but moving the configuration files in subfolders because of this bug.

**Comment by jtpio** [comment]
> This bug is very annoying for non-techy users. I share locked-down jupyter notebooks with my coworkers as a form of rudimentary GUI, and they can't help but moving the configuration files in subfolders because of this bug.

The single-click open was added to Notebook 7 mostly to match the behavior of the classic notebook v6.

Ideally, we should switch back to having the same behavior in both JupyterLab and Notebook at some point. Personally I would be in favor of reverting back to double click to open files in new browser tabs, which would fix this issue. But it's not clear what other users think about this, and whether the prefer the single-click behavior.

**Comment by pythonic2020** [comment]
For what it's worth, I agree with @jtpio on reverting to double-click behavior for Notebook. That sounds like a simpler solution than fixing issues related to the addition of single-click functionality.

**Comment by jtpio** [comment]
If someone would like to open a PR to change the default, this is where is it currently defined:

https://github.com/jupyter/notebook/blob/3edc44b62ac25bb487114c0f1ed757397bc0fb60/packages/tree-extension/src/index.ts#L210

**Comment by jtpio** [comment]
FYI I opened https://github.com/jupyter/notebook/pull/7624 to change the default.

If folks want to try it, you can use the Binder link for the PR: https://mybinder.org/v2/gh/jtpio/notebook/single-click-navigation-default?urlpath=tree

**Comment by jtpio** [comment]
Based on the discussion in https://github.com/jupyter/notebook/pull/7624, it sounded better to look into fixing the actual bug instead of changing the default to double click.

So for those having the issue, the easiest for now would be to go to the settings and change the default:

![Image](https://github.com/user-attachments/assets/383cddb2-accc-4be0-ac56-ead63e799f9d)

Hopefully there will be a fix coming for this in a future 7.4.x release.

**Comment by jtpio** [comment]
Also when looking into this issue I was able to reproduce the issue with Firefox (137), but in Chrome (134). Not sure if that's the case for everyone.

**Comment by jtpio** [comment]
@pythonic2020 @miellaby would you be able to try the changes from https://github.com/jupyter/notebook/pull/7627, to see if it fixes the issue?

For example using the Binder link, so it's easier to try in the browser directly: https://mybinder.org/v2/gh/jtpio/notebook/fix-single-click-navigation?urlpath=tree

**Comment by miellaby** [comment]
> [@pythonic2020](https://github.com/pythonic2020) [@miellaby](https://github.com/miellaby) would you be able to try the changes from [#7627](https://github.com/jupyter/notebook/pull/7627), to see if it fixes the issue?
> 
> For example using the Binder link, so it's easier to try in the browser directly: https://mybinder.org/v2/gh/jtpio/notebook/fix-single-click-navigation?urlpath=tree

The file browser behind this link doesn't exhibit involuntary drag-and-drop actions while being in single-click mode. I validate this part of the correction.

**Comment by pythonic2020** [comment]
I agree with @miellaby that the "involuntary drag-and-drop actions while being in single-click mode" issue appears to have been fixed in the Binder version, but that the app is in single-click mode by default even though the single-click settings box is unchecked, as I wrote above.
